R/core_jobj.R

```diff
@@ -48,14 +48,24 @@ print_jobj <- function(sc, jobj, ...) {
   UseMethod("print_jobj")
 }
 
+get_valid_jobjs <- function(con) {
+  con$state$validJobjs <- con$state$validJobjs %||% new.env(parent = emptyenv())
+  con$state$validJobjs
+}
+
+get_to_remove_jobjs <- function(con) {
+  con$state$toRemoveJobjs <- con$state$toRemoveJobjs %||% new.env(parent = emptyenv())
```

Comment:
@Loquats  Hey sorry my bad... I just realized because this needs to be run on Spark worker nodes, where we don't import `%||%` from the `rlang` namespace, we actually cannot have the `%||%` operator here (and some of your test failures were related to this)

In fact, because methods from `core_*.R` and `worker_*.R` files may be run on Spark worker nodes, we want to ensure functions in those files only depend on constructs from the `base` R package. So in the interest of minimizing R dependencies for worker nodes, we are still better of replacing `x %||% y` with `if (is.null(x)) y` here.
```suggestion
  if (is.null(con$state$toRemoveJobjs))
    con$state$toRemoveJobjs <- new.env(parent = emptyenv())
```

(and same for `validJobjs`)